apps/scratch/src/utilities/scratch-student-activities/senders/send-create-activity.ts

```diff
@@ -0,0 +1,21 @@
+import { ACTIVITY_TYPE } from "../../constants";
+import type {
+  CreateActivityRequest,
+  StudentCreateActivity,
+} from "../../../types/scratch-student-activities";
+
+export async function sendCreateActivity(
+  data: StudentCreateActivity,
+  { sendRequest, action, solution }: CreateActivityRequest,
```

Comment:
I would drop the `CreateActivityRequest` type and either just accept `sendRequest` and `solution` as positional parameters OR define a new type shared between the `sendXActivity` functions bundling `sendRequest` and `solution` but **not** `action`. The `action` should be hardcoded in the respective function, otherwise you could call this function with action = "delete" which does not make sense.